### DISCUSSION TITLE
Bug in hetero_link_pred.py walkthrough

### DISCUSSION THREAD

**Discussion by sperry304** [pyg-team/pytorch_geometric#6666] [Q&A]
Hi there, I'm trying to run the code [here](https://github.com/pyg-team/pytorch_geometric/blob/master/examples/hetero/hetero_link_pred.py) and encountered the following issue at two points in the script - when trying to run the initial `model.encoder` call and then again when trying to train the model. Everything else in this script and the companion CSV loader script seems to work great. Any ideas? 

Here's the error message:
https://pastebin.com/AySBvnrG

I did forget one other error that pops up in the `torch.bincount` call:
`RuntimeError: bincount only supports 1-d non-negative integral inputs.`

Here's the shape of the final training data object:
HeteroData(
  user={ num_nodes=610 },
  movie={ x=[9742, 404] },
  (user, rates, movie)={
    edge_index=[2, 85712],
    edge_label=[85712, 1],
    edge_label_index=[2, 85712]
  },
  (movie, rev_rates, user)={ edge_index=[2, 85712] }
)

NOTE: The error seems to be caused by something from the load CSV script, not this one. Sorry and please feel free to delete.

**Comment by rusty1s** [comment]
Mh, this works fine for me on PyG master. Which version are you using?

**Reply by sperry304** [reply]
> Mh, this works fine for me on PyG master. Which version are you using?

You're right, I was using some slightly differently formatted data files from the load CSV script. This can go ahead and be closed :)
